stl/inc/mutex

```diff
@@ -170,17 +170,8 @@ public:
         _Other._Owns = false;
     }
 
-    unique_lock& operator=(unique_lock&& _Other) noexcept /* strengthened */ {
-        if (this != _STD addressof(_Other)) {
-            if (_Owns) {
-                _Pmtx->unlock();
-            }
-
-            _Pmtx        = _Other._Pmtx;
-            _Owns        = _Other._Owns;
-            _Other._Pmtx = nullptr;
-            _Other._Owns = false;
-        }
+    unique_lock& operator=(unique_lock&& _Other) noexcept {
+        unique_lock(_STD move(_Other)).swap(*this);

```

Comment:
I'm actually not sure about this part. Performance wise I think that the before implementation does less work, and the difference should not be observable.

Reply:
I think that if it had originally been written to Do What The Standard (Now) Says, we wouldn't have considered a dramatic expansion of the code to be worth it.